proxy/start_container_interceptor.go

```diff
@@ -18,25 +18,33 @@ func (i *startContainerInterceptor) InterceptRequest(r *http.Request) error {
 		return err
 	}
 
-	if !containerShouldAttach(container) || r.Header.Get("Content-Type") != "application/json" {
-		i.proxy.createWait(r, container.ID)
-		return nil
-	}
-
-	hostConfig := map[string]interface{}{}
-	if err := unmarshalRequestBody(r, &hostConfig); err != nil {
-		return err
-	}
-
-	i.proxy.addWeaveWaitVolume(hostConfig)
-	if dnsDomain := i.proxy.getDNSDomain(); dnsDomain != "" {
-		if err := i.proxy.setWeaveDNS(hostConfig, container.Config.Hostname, dnsDomain); err != nil {
+	// If the client has sent some JSON which might be a HostConfig, add our
+	// parameters back into it, otherwise Docker will consider them overwritten
+	if containerShouldAttach(container) && r.Header.Get("Content-Type") == "application/json" {
+		params := map[string]interface{}{}
+		if err := unmarshalRequestBody(r, &params); err != nil {
 			return err
 		}
-	}
-
-	if err := marshalRequestBody(r, hostConfig); err != nil {
-		return err
+		// HostConfig can be sent either unnamed at top level or as a struct named HostConfig
+		hostConfig := params
+		if subParam, found := params["HostConfig"]; found {
+			if typecast, ok := subParam.(map[string]interface{}); ok {
+				hostConfig = typecast
+			}
+		}
+		if len(hostConfig) > 0 {
```

Comment:
I think that will leave both pointers as nil so do nothing.  Unless you've tested it?